backend/app/services/subscription/market_access.py

```
@@ -0,0 +1,75 @@
+    db: Session, user_ids: Optional[Iterable[int]]
+) -> List[int]:
+    """Filter whitelist IDs to active users that actually exist."""
+    normalized_ids = normalize_market_whitelist_user_ids(user_ids)
+    if not normalized_ids:
+        return []
+
+    existing_user_ids = {
+        row[0]
+        for row in db.query(User.id).filter(
+            User.id.in_(normalized_ids),
+            User.is_active == True,
```

Ruff (0.15.2)

[error] Avoid equality comparisons to `True`; use `User.is_active:` for truth checks

Replace with `User.is_active`

(E712)